### PR TITLE
Adapt HERON to use ravenframework

### LINKED ISSUE #141: [DEFECT] Adapt HERON to use RAVEN framework as a package
--------
Defect Description
--------

**Describe the defect**

##### What did you expect to see happen?
Changes to RAVEN require we change how we import RAVEN modules. 

##### Do you have a suggested fix for the development team?

**Describe how to Reproduce**
1. 
2. 
3. 
4. 

**Screenshots and Input Files**
Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.

**Platform (please complete the following information):**
 - Version: [e.g. 22]
 - Dependencies Installation: [CONDA or PIP]

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by dylanjm**
--------
Pull Request Description
--------
##### What issue does this change request address?

Closes #141 

##### What are the significant changes in functionality due to this change request?
There should be no changes in functionality. Only adapting code to use a new RAVEN feature. 

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/HERON/wiki/Code-Standards) for details.
- [x] 4. Automated Tests should pass.
- [x] 5. If significant functionality is added, there must be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large tes.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added, the the analytic documentation must be updated/added.
- [x] 9. If any test used as a basis for documentation examples have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: heron [hunk 1]
```diff
@@ -12,8 +12,7 @@ fi
 HERON_DIR=$(cd $HERON_DIRNAME; pwd)
 
 # identify RAVEN location
-RAVENFW_DIR=$(python $HERON_DIRNAME/src/_utils.py get_raven_loc)
-RAVEN_DIR=$(cd $RAVENFW_DIR/../; pwd)
+RAVEN_DIR=$(python $HERON_DIRNAME/src/_utils.py get_raven_loc)
```
Comment (joshua-cogliati-inl) [inline heron:15]: Hm, get_raven_loc is returning: /home/civet/civet/build_0/raven/ravenframework so the above change should probably be reverted, or _utils.py should be modified. It looks like RAVEN_DIR is only used to find establish_conda_env.sh

See output:
```
Starting HERON ...
 ... RAVEN located at "/home/civet/civet/build_0/raven/ravenframework"
 ... Activating CONDA environment ...
/home/civet/civet/build_0/HERON/src/../heron: line 19: /home/civet/civet/build_0/raven/ravenframework/scripts/establish_conda_env.sh: No such file or directory
Running HERON ...
```

Comment (dylanjm) [reply heron:15]: Ahh yes, I just realized I made some manual changes to `.ravenconfig.xml` that would affect this test. I'll make the proper changes.

**Comment by dgarrett622** [comment]
Note that some tests may not pass now due to changes made in RAVEN PR[#1780](https://github.com/idaholab/raven/pull/1780) to percentile and value at risk. The differences should be small, but some tests may need new gold files.

**Comment by dylanjm** [comment]
@PaulTalbot-INL All tests passed and diffs resolved. This is ready for a full review and merge.

**Review by PaulTalbot-INL — APPROVED** [review]

**Comment by PaulTalbot-INL** [comment]
Checklist complete. Thanks for taking care of this, and sorry it took so much effort!

**Comment by PaulTalbot-INL** [comment]
See https://github.com/idaholab/raven/pull/1808 for submod update

**Pull request merged**